src/frontend/src/components/plugins/PluginUIFeatureTypes.ts

```diff
@@ -35,11 +36,7 @@ export type TemplateEditorUIFeature = {
     template_type: ModelType.labeltemplate | ModelType.reporttemplate;
     template_model: ModelType;
   };
-  responseOptions: {
-    key: string;
-    title: string;
-    icon: InvenTreeIconType;
-  };
+  responseOptions: PluginUIFeature;
```

Comment:
@SchrodingersGat sorry, I only had a chance now to review this *properly*.

The idea here was, that each different feature can respond with some different options (which are defined in the "options" dict key provided by the `get_ui_features` function on the plugin class). That's why I build a TS generic type. Where each feature gets its base things defined (`feature_type`, `requestContext`, `responseOptions`, ...). I already defined them for `TemplateEditorUIFeature` (and `TemplatePreviewUIFeature`) here. (Your dashboard things are missing) Then some generic type helpers can be used where needed to type things. This becomes pretty handy, when we publish the inventree-ui npm package. UI feature developers then can just do something like this:

```ts
import { PluginUIGetFeatureType, TemplateEditorUIFeature } from '@inventree/ui';

export const getFeature: PluginUIGetFeatureType<TemplateEditorUIFeature> = (params) => {
  //
};
```  

You now only changed the responseOptions for the template editor feature to some "generic" options (Same in the plugin definition. It's not really clear now what are custom options specific to that feature and what are mandatory things like source. From my two examples this may be not really clear, because they both contained the same options, but I was planning, that other features may need to have different options. E.g. not all features need a description or icon. How do you want to move forward with this now, especially with inventree 1.0?

Reply:
@wolflu05 i"m happy to look at expanding the typing system as part our efforts towards 1.0 :)